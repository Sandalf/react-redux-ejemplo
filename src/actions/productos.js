import * as actionTypes from '../constants/actionTypes';

export function setProductos(producto) {
  return {
    type: actionTypes.PRODUCTOS_SET,
    producto
  };
};


export function removeProducto(indice) {
  return {
    type: actionTypes.PRODUCTOS_REMOVE,
    indice
  };
};