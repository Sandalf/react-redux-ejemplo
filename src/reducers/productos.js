import * as actionTypes from '../constants/actionTypes';

const initialState = ["Manzanas","Leche","Frijoles"];

export default function(state = initialState, action) {
  switch (action.type) {
    case actionTypes.PRODUCTOS_SET:
      return setProductos(state, action);
    case actionTypes.PRODUCTOS_REMOVE:
      return removeProducto(state, action);
    default:
      return state;
  }
}

export function setProductos(state, action) {
  const { producto } = action;
  return [...state,producto];
}

export function removeProducto(state, action) {
  const { indice } = action;
  console.log([...state.slice(0,indice),...state.slice(indice+1)]);
  return [...state.slice(0,indice),...state.slice(indice+1)];
}
