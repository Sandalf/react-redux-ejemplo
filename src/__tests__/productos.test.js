import { removeProducto } from '../reducers/productos.js';

describe('eliminar producto', () => {
  it('deberia eliminar un producto del arreglo', () => {
    let estado = ["Manzana","Tortillas","Leche"];
    let action = { indice: 1 };

    expect(removeProducto(estado, action)).toMatchSnapshot();
  });
});