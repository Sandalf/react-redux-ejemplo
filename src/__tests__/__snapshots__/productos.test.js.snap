// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eliminar producto deberia eliminar un producto del arreglo 1`] = `
Array [
  "Manzana",
  "Leche",
]
`;
